torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -558,6 +571,13 @@ class ProcessGroupNCCL : public ProcessGroup {
   //      "0,4,5,6,7,1,2,3"
   //
   //      Note that the order of the device for the tensor list matters.
+  //
+  // For point-to-point operations:
+  // The key is a string of my current rank and the peer process rank.
+  // e.g. If process 1 and process 2 are involved in a point-to-point communication,
+  // the key will be "1->2" on process 1 and "2->1" on process 2.
```

Comment:
Not sure if I understand the question correctly. The current design only creates 1 communicator per pair of processes but the key is different. When process 1 send/recv to/from process 2, it will use the same communicator. We can sort the ranks so the key will be the same on both processes.